apps/openmw/mwclass/mobile.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef GAME_MWCLASS_MOBILE_H
+#define GAME_MWCLASS_MOBILE_H
+
+#include "../mwworld/class.hpp"
+
+namespace ESM
+{
+    struct GameSetting;
+}
+
+namespace MWClass
+{
+    /// \brief Class holding functionality common to Creature and NPC
+    class Mobile : public MWWorld::Class
+    {
+    protected:
+
+        Mobile();
+
+    public:
+        ~Mobile();
```

Comment:
On Mon, Aug 24, 2015 at 06:27:32AM -0700, Marc Zinnschlag wrote:
>    The destructor is virtual, because the Class destructor is virtual. But
>    yes, for the sake of clarity and consistency the destructor should be
>    marked as virtual. I guess that's what you meant?

Of course it's currently virtual, due to the base class' dtor being virtual. By "future-proofing" I mean marking it virtual so that, if the base class is ever changed/removed, we don't end up with broken dtors.

Trying to save our future selves.